File: client-app/src/app/models/activity.ts
export interface IActivity {
    id: string;
    title: string; 
    description: string;
    category: string;
    date: Date; 
    city: string;
    venue: string;
    isGoing: boolean;   // Note! This refers to the current login User if he is going to the activity
    isHost: boolean;    // Note! This refers to the current login User if he is hosting the activity
    attendees: IAttendee[];
    comments: IComment[];
}

export interface IComment {
    id: string;
    createdAt: Date;
    body: string;
    username: string;
    displayName: string;
    image: string;
}

export interface IActivityFormValues extends Partial<IActivity> {
    time?: Date
}

export class ActivityFormValues implements IActivityFormValues {
    id?: string = undefined;                 
    title: string = '';
    category: string = '';
    description: string = '';
    date?: Date = undefined;   // Note! To be combined with time
    time?: Date = undefined;   // Note! To be combined with date
    city: string = '';
    venue: string = '';
    
    constructor(init?: IActivityFormValues) {
        if (init && init.date) {
            init.time = init.date;
            //Object.assign(this, init);
        }

        Object.assign(this, init);
    }
}

export interface IAttendee {
    username: string;
    displayName: string;
    image: string;
    isHost: boolean;
}

File: Domain/Activity.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Activity
    {
        public Guid Id { get; set; }
        public String Title { get; set; }
        public String Description { get; set; }
        public String Category { get; set; }
        public DateTime Date { get; set; }
        public String City { get; set; }
        public String Venue { get; set; }
        public virtual ICollection<UserActivity> UserActivities { get; set; } // Note! virtual is essential for lazy loading
        public virtual ICollection<Comment> Comments { get; set; }
    }
}